doc/userguide/rules/intro.rst

```diff
@@ -349,3 +349,201 @@ reassembled streams, TLS-, SSL-, SSH-, FTP- and dcerpc-buffers.
 
 Note that there are some exceptions, e.g. the ``http_raw_uri`` keyword.
 See :ref:`rules-http-uri-normalization` for more information.
+
+
+Rule's Types and Categorization
+-------------------------------
+
+Once parsed, Suricata rules are categorized for performance and further
+processing (as different rule types will be handled by specific engine modules).
+The signature types are defined in `src/detect.h
+<https://github.com/OISF/suricata/blob/master/src/detect.h#L62-L75>`_.
+
+The rule type will impact:
+
+  - Scope
```

Comment:
can this `scope` be defined more precisely?

Reply:
I think I'm seeing the scope very closely related to the Rule properties, or what does the rule action apply to. I'll try to elaborate on that, here.